server/src/test/java/org/opensearch/action/search/SearchRequestTests.java

```diff
@@ -238,6 +243,146 @@ public void testValidate() throws IOException {
+        }
+        {
+            // _shard_doc + search_after but NO PIT -> reject (explicitly exercising this combo)
+            SearchRequest searchRequest = new SearchRequest().source(
+                new SearchSourceBuilder().sort(SortBuilders.shardDocSort()).searchAfter(new Object[] { 1L })
+            );
+            ActionRequestValidationException e = searchRequest.validate();
+            assertNotNull(e);
+            assertEquals(1, e.validationErrors().size());
+            assertEquals(
+                "_shard_doc is only supported with point-in-time (PIT). Add a PIT or remove _shard_doc.",
+                e.validationErrors().get(0)
+            );
+        }
+        {
+            // Error aggregation with SCROLL: collect multiple errors including shard_doc+scroll
+            SearchRequest searchRequest = new SearchRequest().source(
+                new SearchSourceBuilder().pointInTimeBuilder(new PointInTimeBuilder("id"))  // PIT + scroll already illegal
+                    .sort(SortBuilders.shardDocSort())                    // shard_doc + scroll = illegal
+                    .from(10)                                             // also illegal with scroll
+                    .size(0)                                              // also illegal with scroll
+            ).scroll(TimeValue.timeValueSeconds(30));
+
+            ActionRequestValidationException e = searchRequest.validate();
+            assertNotNull(e);
+            assertThat(
+                e.validationErrors(),
+                hasItems(
+                    "using [point in time] is not allowed in a scroll context",
+                    "_shard_doc cannot be used with scroll. Use PIT + search_after instead.",
+                    "using [from] is not allowed in a scroll context",
+                    "[size] cannot be [0] in a scroll context"
+                )
+            );
+        }
+        {
+            // PIT present + search_after but NO _shard_doc -> valid (control for PIT/search_after)
+            SearchRequest searchRequest = new SearchRequest().source(
+                new SearchSourceBuilder().pointInTimeBuilder(new PointInTimeBuilder("id"))
+                    .sort("_id", SortOrder.ASC)
+                    .searchAfter(new Object[] { "x" })
+            );
+            ActionRequestValidationException e = searchRequest.validate();
+            assertNull(e);
+        }
+        {
+            // FieldSortBuilder("_shard_doc") WITH PIT -> valid (object-style declared as FieldSort)
+            SearchRequest searchRequest = new SearchRequest().source(
+                new SearchSourceBuilder().pointInTimeBuilder(new PointInTimeBuilder("id"))
+                    .sort(new FieldSortBuilder(ShardDocSortBuilder.NAME).order(SortOrder.ASC))
+            );
+            ActionRequestValidationException e = searchRequest.validate();
+            assertNull(e);
```

Comment:
Have you try `new FieldSortBuilder(ShardDocSortBuilder.NAME).order(SortOrder.ASC)`? it will fail with
```
"No mapping found for [" + fieldName + "] in order to sort on"
```